src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1650,8 +1758,29 @@ protected override BoundNode VisitReturnStatementNoAdjust(BoundReturnStatement n
             }
 
             EnforceDoesNotReturn(node.Syntax);
```

Comment:
>EnforceDoesNotReturn [](start = 12, length = 20)

This is also misplaced.  An infinite loop or throw statement inside a finally should prevent the diagnostic reported by this call.